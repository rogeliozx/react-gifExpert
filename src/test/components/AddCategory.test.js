import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import AddCategory from './../../components/AddCategory';

describe('Test AddCategory', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  //esto se llama antes de cada prueba
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test('should show render rigth', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should change textInput', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    input.simulate('change', { target: { value } });
    const valueInput = wrapper.find('input').get(0).props.value;
    expect(valueInput).toBe(value);
  });

  test('should no post the info whe is submited', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
  test('should call setCategories and cleant text box', () => {
    const value = 'Hola mundo';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
