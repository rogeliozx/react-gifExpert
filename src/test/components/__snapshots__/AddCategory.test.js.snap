// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddCategory should show render rigth 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add Category
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
